test/tbb/test_collaborative_call_once.cpp

```diff
@@ -0,0 +1,372 @@
+#if __GNUC__ && !__TBB_GCC_PARAMETER_PACK_IN_LAMBDAS_PRESENT
+            tbb::collaborative_call_once(flag, func);
+#else
+            tbb::collaborative_call_once(flag, body, std::forward<Args>(args)...);
+#endif // __TBB_GCC_PARAMETER_PACK_IN_LAMBDAS_PRESENT
+        }
+    });
+}
+
+template<typename Fn, typename... Args>
+void call_once_threads(Fn& body, Args&&... args) {
+    tbb::collaborative_once_flag flag;
+    std::vector<std::thread> threads;
+
+#if __GNUC__ && !__TBB_GCC_PARAMETER_PACK_IN_LAMBDAS_PRESENT
+    auto stored_pack = tbb::detail::d0::save_pack(std::forward<Args>(args)...);
+    auto func = [&] { tbb::detail::d0::call(body, stored_pack); };
+#endif // !__TBB_GCC_PARAMETER_PACK_IN_LAMBDAS_PRESENT
+
+    for (int i = 0; i < 100; ++i)
+    {
+        threads.push_back(std::thread([&]() {
+#if __GNUC__ && !__TBB_GCC_PARAMETER_PACK_IN_LAMBDAS_PRESENT
+            tbb::collaborative_call_once(flag, func);
+#else
+            tbb::collaborative_call_once(flag, body, std::forward<Args>(args)...);
+#endif // __TBB_GCC_PARAMETER_PACK_IN_LAMBDAS_PRESENT
+        }));
+    }
+    for (auto& thread : threads) {
+        thread.join();
+    }
+}
+
+//! Test for functor to be called only once
+//! \brief \ref interface \ref requirement
+TEST_CASE("only calls once 1") {
+    {
+        increment_functor f;
+
+        call_once_in_for_loop(f);
+
+        REQUIRE( f.ct == 1);
+    }
+    {
+        increment_functor f;
+
+        call_once_in_parallel_for(f);
+
+        REQUIRE(f.ct == 1);
+    }
+    {
+        increment_functor f;
+
+        call_once_threads(f);
+
+        REQUIRE(f.ct == 1);
+    }
+}
+
+//! Test for functor to be called only once
+//! \brief \ref interface \ref requirement
+TEST_CASE("only calls once 2") {
+    {
+        sum_functor f;
+
+        call_once_in_for_loop(f, 1, 2, 3 ,4);
+
+        REQUIRE(f.sum == 10);
+    }
+    {
+        sum_functor f;
+
+        call_once_in_parallel_for(f, 1000, -1000);
+
+        REQUIRE(f.sum == 0);
+    }
+    {
+        sum_functor f;
+
+        call_once_threads(f, 0, -1, -5);
+
+        REQUIRE(f.sum == -6);
+    }
+}
+
+//! Stress test for functor to be called only once
+//! \brief \ref interface \ref requirement \ref stress
+TEST_CASE("only calls once - stress test") {
+    {
+        increment_functor f;
+
+        tbb::collaborative_once_flag flag;
+        std::vector<std::thread> threads;
+        for (std::size_t i = 0; i < utils::get_platform_max_threads()*5; ++i)
+        {
+            threads.push_back(std::thread([&]() {
+                tbb::collaborative_call_once(flag, f);
+            }));
+        }
+        for (auto& thread : threads) {
+            thread.join();
+        }
+        REQUIRE(f.ct == 1);
+    }
+    {
+        increment_functor f;
+
+        utils::SpinBarrier barrier(utils::get_platform_max_threads()*5);
+        tbb::collaborative_once_flag flag;
+        utils::NativeParallelFor(utils::get_platform_max_threads()*5, [&](std::size_t) {
```

Comment:
We want to check cases when there are more threads than `max_nfs_size`. Does it make sense to rely on `TBB_TEST_LOW_WORKLOAD` for reduced workload but in general case create many threads, e.g. `4*max_nfs_size`?